zarr/codecs.py

```diff
@@ -485,6 +485,105 @@ def __repr__(self):
 codec_registry[Delta.codec_id] = Delta
 
 
+class AsType(Codec):
+    """Filter to convert data between different types.
+
+    Parameters
+    ----------
+    dtype : dtype
+        Data type used to store the data.
+    astype : dtype, optional
+        Data type of the data read in.
```

Comment:
TBH when I saw `Delta`. The naming felt unclear and a bit confusing. That being said, I completely agree that having this opposite to `Delta` doesn't help either. Personally, I'd rather change the names of these to something else entirely like `from_dtype` and `to_dtype` or perhaps `data_dtype` and `view_dtype`. Though any names that make it clear what these dtypes are applied seem ok IMHO. What do you think?